src/App.tsx

```diff
@@ -1,116 +1,81 @@
 import React from 'react';
-import style from './App.module.scss';
-import Header from './components/Header/Header';
-import CardItems from './components/CardItems/CardItems';
-import { URL } from './models/enums';
+import { ROUTER_PATHS, URL } from './models/enums';
 import { ShowData } from './models/interfaces';
-import { MdReportGmailerrorred } from 'react-icons/md';
-import ErrorBoundary from './components/ErrorBoundary/ErrorBoundary';
-
-interface AppProps {
-  shows: ShowData[];
-  isLoading: boolean;
-  error: Error | null | unknown;
-  currentPage: number;
-  itemsPerPage: number;
-  isMoreShows: boolean;
-  searchQuery: string;
-  isShowMoreButtonDisable: boolean;
-  isClickedErrorButton: boolean;
-}
-class App extends React.Component<object, AppProps> {
-  constructor(props: AppProps) {
-    super(props);
-    this.state = {
-      shows: [],
-      isLoading: false,
-      error: null,
-      currentPage: 0,
-      itemsPerPage: 20,
-      isMoreShows: false,
-      searchQuery: '',
-      isShowMoreButtonDisable: false,
-      isClickedErrorButton: false,
-    };
-
-    this.loadShows = this.loadShows.bind(this);
-    this.showMoreShows = this.showMoreShows.bind(this);
-    this.handleSearch = this.handleSearch.bind(this);
-    this.enableButton = this.enableButton.bind(this);
-    this.throwError = this.throwError.bind(this);
-  }
-
-  async componentDidMount() {
-    this.loadShows();
-    this.enableButton();
-  }
-
-  async loadShows() {
-    const { currentPage, isLoading } = this.state;
-
+import { Route, Routes, useNavigate } from 'react-router-dom';
+import Layout from './components/Layout/Layout';
+import MainPage from './components/pages/MainPage/MainPage';
+import NotFound from './components/pages/NotFound/NotFound';
+import DetailsPage from './components/pages/DetailsPage/DetailsPage';
+
+const App: React.FC = () => {
+  const [shows, setShows] = React.useState<ShowData[]>([]);
```

Comment:
It's more convenient to import and use hooks without React (using destructuring):
```import { useState } from 'react'```

Reply:
Done